File: 219-ContainsDuplicateIi/219-ContainsDuplicateIi.java
// Last updated: 5/29/2026, 11:57:16 AM
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer,Integer> map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(map.containsKey(nums[i])){
                if(Math.abs(i-map.get(nums[i]))<=k){
                    return true;
                }
            }
            map.put(nums[i],i);
        }
        return false;
    }
}

File: 219-ContainsDuplicateIi/219-ContainsDuplicateIi.py
# Last updated: 5/29/2026, 11:57:10 AM
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        map=defaultdict(int)
        for i in range(len(nums)):
            if nums[i] in map:
                if abs(i-map[nums[i]])<=k:
                    return True
            map[nums[i]]=i
        return False